src/win/util.c

```diff
@@ -484,7 +484,7 @@ uint64_t uv__hrtime(double scale) {
    * performance counter interval, integer math could cause this computation
    * to overflow. Therefore we resort to floating point math.
    */
-  return (uint64_t) ((double) counter.QuadPart * hrtime_interval_ * scale);
+  return (uint64_t) (counter.QuadPart * (uint64_t) scale / hrtime_interval_);
```

Comment:
But the real question is whether `counter.QuadPart * scale` won't overflow. I think this is a real risk. I've values for `scale` up to `1e9` which (if my calculation is correct) would make the high-resulution timer wrap after windows is up for about 200 days. That doesn't seem too likely but it's definitely in the zone where I'm uncomfortable with it.